src/components/utils/integration_tests/onfidoApi.test.js

```diff
@@ -0,0 +1,731 @@
+import {
+  uploadDocument,
+  uploadLivePhoto,
+  uploadSnapshot,
+  sendMultiframeSelfie,
+  requestChallenges,
+  uploadLiveVideo,
+} from '../onfidoApi'
+import fs from 'fs'
+
+const API_URL = 'https://api.onfido.com'
+let jwtToken = null
+const getTestJwtToken = (resolve) => {
+  const tokenFactoryUrl = 'https://token-factory.onfido.com/sdk_token' // EU region
+  const request = new XMLHttpRequest()
+  request.open('GET', tokenFactoryUrl, true)
+  request.setRequestHeader(
+    'Authorization',
+    `BASIC ${process.env.SDK_TOKEN_FACTORY_SECRET}`
+  )
+  request.onload = function () {
+    if (request.status >= 200 && request.status < 400) {
+      const data = JSON.parse(request.responseText)
+      return resolve(data.message)
+    }
+  }
+  request.send()
+}
+
+const EXPIRED_JWT_TOKEN =
+  'eyJhbGciOiJIUzI1NiJ9.eyJleHAiOjE2MDc3MDc1NTEsInBheWxvYWQiOiJycUFvMEtSbXdtWlViWFRLUHp2TXlTaGZtelNDNVhtRWM3aVZ4ZzJ5b2NQbEQrbk9rQmxtcHBaK0FCKzBcbkwveEtYRm4yeTBNZGxNNXRXVE5HeVNVSG5nPT1cbiIsInV1aWQiOiJpd29rRlZlZEcxOCIsImVudGVycHJpc2VfZmVhdHVyZXMiOnsiY29icmFuZCI6dHJ1ZSwiaGlkZU9uZmlkb0xvZ28iOnRydWV9LCJ1cmxzIjp7InRlbGVwaG9ueV91cmwiOiJodHRwczovL3RlbGVwaG9ueS5vbmZpZG8uY29tIiwiZGV0ZWN0X2RvY3VtZW50X3VybCI6Imh0dHBzOi8vc2RrLm9uZmlkby5jb20iLCJzeW5jX3VybCI6Imh0dHBzOi8vc3luYy5vbmZpZG8uY29tIiwiYXV0aF91cmwiOiJodHRwczovL2VkZ2UuYXBpLm9uZmlkby5jb20iLCJvbmZpZG9fYXBpX3VybCI6Imh0dHBzOi8vYXBpLm9uZmlkby5jb20iLCJob3N0ZWRfc2RrX3VybCI6Imh0dHBzOi8veGQub25maWRvLmNvbSJ9fQ.Ece4NQpZLsPzsgd6W4kDYNugW66W_Fl__jfz6d96WEI'
+const EXPECTED_EXPIRED_TOKEN_ERROR = {
+  status: 401,
+  response: {
+    error: {
+      fields: {},
+      message: 'The token has expired, please request a new one',
+      type: 'expired_token',
+    },
+  },
+}
+
+const DOCUMENT_VALIDATIONS = {
+  detect_document: 'error',
+  detect_cutoff: 'warn',
+  detect_glare: 'warn',
+  detect_blur: 'warn',
+}
+
+const TEST_DOCUMENT_DATA = {
+  sdkMetadata: {},
+  validations: { ...DOCUMENT_VALIDATIONS },
+  side: 'front',
+  type: 'passport',
+}
+
+const TEST_VIDEO_DATA = {
+  challengeData: {
+    challenges: [
+      {
+        query: [7, 8, 0],
+        type: 'recite',
+      },
+      {
+        query: 'turnRight',
+        type: 'movement',
+      },
+    ],
+    id: 'test-challenge-data',
+    switchSeconds: 2021,
+  },
+  language: 'en_US',
+  sdkMetadata: {},
+}
+
+const PATH_TO_TEST_FILES = `${__dirname}/./../../../../test/resources/`
+
+const createEmptyFile = (
+  testFileName = 'empty_file.jpg',
+  mimeType = 'image/jpeg'
+) =>
+  new File([], testFileName, {
+    type: mimeType,
+  })
+
+/* eslint jest/no-test-callback: 0 */
```

Comment:
I understood the root cause of why we have to surpass this ESLint rule here, but I want to extend the idea a bit:

- Because the methods of `onfidoApi.js` were implemented using callbacks, it has several limitations like callbacks hell, unnecessary complexity, couldn't take advantages of Promises, etc. Hence I strongly suggest that we have another ticket to convert those calls into Promise, so we can have a modern (and better) approach on asynchronous codes.
- We can use `fs.readFileSync()` to avoid callbacks like in `fs.readFile()`

Reply:
Yes, agree that there there should be a follow up ticket to address the 1st point here. 
Will address the 2nd point in this PR first alongside the refactor to break `onfidoApi.test.js` into smaller chunks for now.